pyfr/solvers/baseadvecdiff/system.py

```diff
@@ -18,6 +18,24 @@ def deps(dk, *names): return self._kdeps(k, dk, *names)
         # Interpolate the solution to the flux points
         g1.add_all(k['eles/disu'])
 
+        # If entropy filtering, compute entropy bounds
+        if 'eles/local_entropy' in k:
+            g1.add_mpi_reqs(m['ent_fpts_recv'])
```

Comment:
Move this block down to get additional overlap.

Reply:
Down to where?

Reply:
Couple of lines so it is after:

```
# Pack and send these interpolated solutions to our neighbours
        g1.add_all(k['mpiint/scal_fpts_pack'], deps=k['eles/disu'])
        for send, pack in zip(m['scal_fpts_send'], k['mpiint/scal_fpts_pack']):
```

At the moment we interpolate the solution to the flux points and so are basically ready to pack them up and send them to our neighbours (which is good as it gets MPI requests started ASAP).  However, in the current graph we instead defer it until after the entropy stuff.  On CUDA and OpenCL this is fine as the runtimes will sort this out, but on HIP (due to bugs) and OpenMP kernels are executed in-order.  Thus we'll end up starting the MPI transfers for the flux points later than we could have.